hack/unwanted-dependencies.json

```diff
@@ -208,8 +168,8 @@
         "sigs.k8s.io/kustomize/api",
         "sigs.k8s.io/kustomize/kustomize/v5"
       ],
-      "github.com/gorilla/mux": [
-        "github.com/Microsoft/hcsshim"
+      "go.opencensus.io": [
```

Comment:
This is interesting… is hnslib depending on this in a way hcsshim wasn't?

Reply:
Are you asking about opencensus? https://github.com/microsoft/hnslib/blob/main/go.mod . its being used for tracing 
![image](https://github.com/user-attachments/assets/06cec304-ea10-4ef7-888c-19bb3487a895)

Reply:
I thought this was a straight code extraction from hcsshim, but opencensus didn't show up as being used by hcsshim  🤔

Reply:
It is used in https://github.com/search?q=repo%3Amicrosoft%2Fhcsshim+%2Fopencensus%2F+language%3A%22Go+Module%22&type=code&l=Go+Module and we have it in our pipeline to move hcsshim to otel. hnslib is using opencensus to keep it identical to hcsshim IMO - cc @princepereira  to confirm

Reply:
> hnslib is using opencensus to keep it identical to hcsshim IMO

that's weird, hcsshim didn't trigger recording this dependency before...

Reply:
Hmm if I remember correctly, there was an entry with hcsshim for opencensus that I removed

Reply:
> I thought this was a straight code extraction from hcsshim, but opencensus didn't show up as being used by hcsshim 🤔

Yes, your understanding is correct. Hnslib is a straight code extraction from hcsshim. Even earlier version of hcsshim (v0.8.26) which k8s was referring to was also using opencensus.

https://github.com/kubernetes/kubernetes/blob/3f7a50f38688eb332e2a1b013678c6435d539ae6/go.mod#L228

Reply:
thanks!

Reply:
got it, our hcsshim version as of a couple weeks ago referenced this: https://github.com/microsoft/hcsshim/blob/v0.8.26/go.mod#L21

it since moved to otel, and hnslib intends to do the same eventually

sgtm